Integrations/integration-VirusTotal-Private_API.yml

```diff
@@ -549,43 +556,103 @@ script:
         """
         corresponds to 'vt-get-url-report' command. Retrieves a report about a url
         """
-
+        global SCAN_FINISH_TIME_IN_SECONDS
         args = demisto.args()
-        url = args.get('resource')
+        urls = argToList(args.get('resource'))
         all_info = 1 if args.get('allInfo', None) == 'true' else 0
-        threshold = int(args.get('threshold', None) or demisto.params().get('urlThreshold', None) or 10);
         short_format = args.get('shortFormat', None) == 'true'
+        retries = int(args.get('retries', 2))
         full_response = args.get('fullResponse', None) == 'true'
+        threshold = int(args.get('threshold', None) or demisto.params().get('urlThreshold', None) or 10)
+        SCAN_FINISH_TIME_IN_SECONDS = int(args.get('retry_time', SCAN_FINISH_TIME_IN_SECONDS))
         if (full_response):
             max_len = 1000
         else:
             max_len = 50
 
-        response = get_url_report(url, all_info)
+        responses_dict = get_url_reports_with_retries(urls, all_info, retries)
+        md = ''
+        ec = {
+                'DBotScore': [],
+                outputPaths['url']: [],
+            }
+        for url, res in responses_dict.iteritems():
+            url_md, url_ec = create_url_report_output(url, res, ec, threshold, max_len, short_format)
```

Comment:
this is very confusing.
`url_ec` is bing returned, but never used, and `ec` is passed as an argument but being manipulated.
my idea was to crate a function that will return all the products of handling one url res. than use this products to build the the larger context and human readable.
`ec` should not be passed to this function but rather be updated by its products.
if you need to return the dotscore separately from the url_ec, that is ok, just return another object,
url_md, url_ec, dbot_score = create_url_report_output(...)